crates/pallet-subspace/src/lib.rs

```diff
@@ -910,9 +956,11 @@ impl<T: Config> Pallet<T> {
         Ok(Pays::No.into())
     }
 
-    fn do_store_root_block(root_block: RootBlock) -> DispatchResult {
-        RecordsRoot::<T>::insert(root_block.segment_index(), root_block.records_root());
-        Self::deposit_event(Event::RootBlockStored(root_block));
+    fn do_store_root_blocks(root_blocks: Vec<RootBlock>) -> DispatchResult {
+        for root_block in root_blocks {
+            RecordsRoot::<T>::insert(root_block.segment_index(), root_block.records_root());
+            Self::deposit_event(Event::RootBlockStored(root_block));
```

Comment:
Not sure if it could be better to emit only one event, e.g., `Self::deposit_event(Event::RootBlocksStored(root_blocks));`

Reply:
I though about that and checked what Polkadot does. Extrinsics like `paraInherent.enter` contain a bunch of things, but produce a set of separate `paraInclusion.CandidateIncluded` and `paraInclusion.CandidateBacked` events as the result, I just followed that analogy. It might also remove some nested loop on client side when listening and iterating over events.